oraclepaas/resource_mysql_access_rule.go

```diff
@@ -0,0 +1,220 @@
+package oraclepaas
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	opcClient "github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/go-oracle-terraform/mysql"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceOraclePAASMySQLAccessRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOraclePAASMySQLAccessRuleCreate,
+		Read:   resourceOraclePAASMySQLAccessRuleRead,
+		Update: resourceOraclePAASMySQLAccessRuleUpdate,
+		Delete: resourceOraclePAASMySQLAccessRuleDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"service_instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"destination": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ports": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			// name validation: start with a letter, include letter, number and hyphen only
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				//ValidateFunc: validateAccessRuleName,
```

Comment:
Fixed. I added the validator class and re-enabled the validation. Currently whats bugging me (slightly) is that that the validation rules on the REST API side may change and it will cause our validation rules to be wrong.